bears/c_languages/AStyleLintBear.py

```diff
@@ -0,0 +1,54 @@
+from coalib.bearlib.abstractions.Linter import linter
+from coalib.bearlib.spacing.SpacingHelper import SpacingHelper
+from dependency_management.requirements.DistributionRequirement import (
+    DistributionRequirement)
+
+
+@linter(executable='astyle',
+        output_format='corrected',
+        use_stdin=True)
+class AStyleLintBear:
+    """
+    Artistic Style is a source code indenter, formatter,
+    and beautifier for the C, C++, C++/CLI, Objective‑C,
+    C# and Java programming languages.
+
+    For more information, consult <http://astyle.sourceforge.net/astyle.html>.
+    """
+
+    LANGUAGES = {'C, C++, Objective-C, C#, Java'}
+    REQUIREMENTS = {DistributionRequirement(apt_get='astyle')}
+    AUTHORS = {'The coala developers'}
+    AUTHORS_EMAILS = {'coala-devel@googlegroups.com'}
+    LICENSE = 'AGPL-3.0'
+    CAN_DETECT = {'Formatting'}
+    CAN_FIX = {'Formatting'}
+
+    @staticmethod
+    def create_arguments(filename, file, config_file,
+                         astyle_style: str='',
+                         astyle_indent: str='',
+                         astyle_astylerc: str='',
+                         astyle_indent_size: int=0):
+        """
+        :param astyle_style:
+            Defines the bracket style to use. If no bracket style is
+            specified, the default 'allman' is used.
+        :param astyle_indent:
+            The indentation type to use. It can either be spaces or tabs.
+        :param astyle_astylerc:
+            Optional configuration file for astyle to read configs.
+        :param astyle_indent_size:
+            Number of spaces per indentation level.
+        """
+        args = ('--dry-run', '--suffix=none')
+        if astyle_style:
+            args += ('--style=' + astyle_style,)
+        if astyle_indent:
+            indent = '--indent=' + astyle_indent
+            indent += '=' + \
+                str(astyle_indent_size) if astyle_indent_size else ''
+            args += (indent,)
+        if astyle_astylerc:
+            args += ('--options=' + astyle_astylerc,)
```

Comment:
actually you can use a list instead of a tuple for `args`, linter doesn't alter the args returned + you don't need them anyway after that call. Makes stuff slightly faster, as appending to tuples copies the old tuple every time^^